Data.Connector.Svc/DependencyExtensions.cs

```diff
@@ -47,11 +48,13 @@ public static IServiceCollection AddDependencies(
     {
         services.AddSingleton<SqlClientFactory>(SqlClientFactory.Instance);
 
+        services.AddSingleton<ApplicationContext>();
```

Comment:
Not sure if we need to add this singleton dependency here, if only we use it once later in this method; we could just instantiate it there.

Reply:
I see what you mean, having the instantiation in section at line 57 instead and then pass on that instance to any item in the iteration. The reason I added it with a Singleton was a note in the SDK itself that this should only ever be one instance of ApplicationContext per application - https://github.com/Azure/iot-operations-sdks/blob/2b7dacd127700e7f8a85dccb52d08980d2ca364d/dotnet/src/Azure.Iot.Operations.Protocol/ApplicationContext.cs#L11